src/DataAPI.jl

```diff
@@ -404,6 +426,42 @@ If `style=true` return `(default, :default)`.
 """
 function colmetadata end
 
+"""
+    colmetadata(x, [col]; style::Bool=false)
+
+Return a dictionary mapping all column metadata keys to metadata values
+associated with column `col` of table `x`. Throw an error if `x` does not
+support reading metadata for column `col` or column `col` is not present in `x`.
+
+If `style=true` values are tuples of metadata value and metadata style. Metadata
+style is an additional information about the kind of metadata that is stored for
+the `key`.
+
+$STYLE_INFO
+
+If `col` is not passed return a dictionary mapping columns represented as
+`Symbol` that have associated metadata to dictionaries dictionary mapping all
```

Comment:
Maybe put this in the first paragraph, as it completely changes the behavior of the function? (It could amost make sense to have a separate docstring but that would be quite heavy...)

Reply:
OK - done.